### PR TITLE
Omnibox Integration

### PR DESCRIPTION
**Pull request by fxnoob**
Omnibox (https://developer.chrome.com/extensions/omnibox) integration.
Steps: 
 -  Type `short` in search bar then hit `TAB` key.
 -  Type any short cut without `s/` prefix in it and then hit `Enter`.

### CODE REVIEW

File: src/background.ts [hunk 1]
```diff
@@ -71,5 +71,14 @@ class ShortExt {
 const webUi = 'https://s.time4hacks.com';
 const apiBaseUrl = `${webUi}/r/`;
 
+// This event is fired with the user accepts the input in the omnibox.
```
Comment (naturetime10) [inline src/background.ts]:
```suggestion
// This event is triggered when the user accepts the input in the omnibox.
```

File: manifest.json [hunk 2]
```diff
@@ -15,9 +15,10 @@
     "webRequest",
     "webRequestBlocking"
   ],
+  "omnibox": { "keyword" : "short" },
```
Comment (naturetime10) [inline manifest.json]:
```suggestion
  "omnibox": { 
    "keyword" : "short"
  },
```

**Comment by byliuyang** [comment]
https://github.com/fxnoob/short-ext/blob/06735bb2626dcbd5898ab14580b80d7cce363b07/src/background.ts#L76-L80

**Comment by naturetime10** [comment]
Original:
```typescript
chrome.omnibox.onInputEntered.addListener(
function(text) {
    // Encode user input for special characters , / ? : @ & = + $ #
    var newURL = 'https://s.time4hacks.com/r/' + encodeURIComponent(text);
    chrome.tabs.create({ url: newURL });
});
```

Suggested:
```typescript
function openTab(url: string) {
    chrome
      .tabs
      .create({
        url: url
      });
}

class ShortExt {
    constructor(private apiBaseUrl: string, private webUi: string) {
    }

    fullURL(alias: string) {
        // Escape user input for special characters , / ? : @ & = + $ #
        let escapedAlias = encodeURIComponent(alias);
        return `${this.apiBaseUrl}${escapedAlias}`;
    }

    setupOmnibox(alias: string) {
        let url = this.fullURL(alias);
        chrome
            .omnibox
            .onInputEntered
            .addListener(()  =>  {
                openTab(url);
            });
    }

    // Keep the rest of the code here
}
```

**Comment by naturetime10** [comment]
Please add a screenshot for preview purpose.

**Review by naturetime10 — CHANGES_REQUESTED** [review]
Thank you very much for working on this feature!
Please address the commends in the pull request.

File: src/background.ts [hunk 3]
```diff
@@ -23,8 +23,34 @@ function extractAlias(shortLink: string): string {
...
+        });
+}
+
 class ShortExt {
     constructor(private apiBaseUrl: string, private webUi: string) {
+        this.fullURL = this.fullURL.bind(this);
+        this.setupOmnibox = this.setupOmnibox.bind(this);
+    }
+
+    fullURL(alias: string) {
```
Comment (naturetime10) [inline src/background.ts]: I remember using 
```typescript
fullURL = (alias: string) => {
};
```
can solve `this` binding issue. Please let me know if this doesn't work.

File: src/background.ts [hunk 4]
```diff
@@ -23,8 +23,34 @@ function extractAlias(shortLink: string): string {
...
+        // Escape user input for special characters , / ? : @ & = + $ #
+        let escapedAlias = encodeURIComponent(alias);
+        return `${this.apiBaseUrl}${escapedAlias}`;
+    }
+
+    setupOmnibox(alias: string) {
```
Comment (naturetime10) [inline src/background.ts]: Sorry for the confusion, it actually should be the following:
```typescript
setupOmnibox = () => {
    chrome
        .omnibox
        .onInputEntered
        .addListener((alias: string)  =>  {
          let url = this.fullURL(alias);
          openTab(url);
        });
    }
}
```

File: src/background.ts [hunk 5]
```diff
@@ -71,14 +97,12 @@ class ShortExt {
...
-// This event is fired with the user accepts the input in the omnibox.
-chrome.omnibox.onInputEntered.addListener(
-function(text) {
-    // Encode user input for special characters , / ? : @ & = + $ #
-    var newURL = 'https://s.time4hacks.com/r/' + encodeURIComponent(text);
-    chrome.tabs.create({ url: newURL });
-});
+const ext = new ShortExt(apiBaseUrl, webUi);
 
+// This event is triggered when the user accepts the input in the omnibox.
```
Comment (naturetime10) [inline src/background.ts]: line 102 - 106 should be removed.

File: src/background.ts [hunk 6]
```diff
@@ -23,8 +23,34 @@ function extractAlias(shortLink: string): string {
...
+    chrome
+        .tabs
+        .create({
+            url: url
+        });
+}
+
 class ShortExt {
     constructor(private apiBaseUrl: string, private webUi: string) {
+        this.fullURL = this.fullURL.bind(this);
```
Comment (naturetime10) [inline src/background.ts]: This should be removed.

File: src/background.ts [hunk 7]
```diff
@@ -23,8 +23,34 @@ function extractAlias(shortLink: string): string {
...
+        this.setupOmnibox = this.setupOmnibox.bind(this);
```
Comment (naturetime10) [inline src/background.ts]: Please replace this with
```typescript
this.setupOmnibox();
```

**Review by naturetime10 — CHANGES_REQUESTED** [review]

**Review by naturetime10 — CHANGES_REQUESTED** [review]
One last thing to change in `manifest.json`:

Please increment version number to 0.0.3
```json
  "version": "0.0.2" 
```

**Review by naturetime10 — APPROVED** [review]

**Pull request merged**